themes/bootstrap3/templates/search/facet-list.phtml

```diff
@@ -28,38 +26,43 @@
     <label><?=$this->translate('Sort') ?></label>
     <div class="btn-group">
       <?php foreach ($this->sortOptions as $key => $sort): ?>
-        <a href="<?=$urlBase . '&amp;facetpage=1&amp;facetsort=' . urlencode($key) ?>" class="btn btn-default js-facet-sort<?php if ($this->sort == $key): ?> active<?php endif; ?>" data-sort="<?=$key ?>" data-lightbox-ignore><?=$this->translate($sort) ?></a>
+        <a href="<?=$this->escapeHtmlAttr($urlBase . '&facetpage=1&facetsort=' . urlencode($key) . '&contains=' . urlencode($contains)) ?>" class="btn btn-default js-facet-sort<?php if($this->sort == $key): ?> active<?php endif; ?>" data-sort="<?=$key ?>" data-lightbox-ignore><?=$this->translate($sort) ?></a>
       <?php endforeach; ?>
     </div>
+
+    <div class="btn-group facet-lightbox-filter hidden">
+      <form>
+        <fieldset>
+          <?php if ($this->results->getParams()->supportsFacetFiltering($this->facet) === true): ?>
+            <legend><?=$this->transEsc('Filter') ?></legend>
+            <input class="ajax_param" data-name="contains" class="form-control" type="text" value="<?=htmlspecialchars($this->contains)?>" aria-label="<?=$this->transEscAttr('search_terms')?>">
+            <button id="btn-reset-contains" <?=strlen($this->contains) < 1 ? 'class="hidden" ' : ''?>type="reset" role="button" aria-label="<?=$this->transEsc('searchform_reset_button')?>"><?=$this->icon('ui-reset-search')?></button>
```

Comment:
The icon is not quite vertically centered for me (Firefox, macOS). I believe the button also needs styling to fit other `form-control` elements.

Reply:
Looks much better for me in Windows after the latest changes; I suspect it will look better for @EreMaijala too now.